client/allocrunner/alloc_runner_hooks.go

```diff
@@ -77,7 +77,8 @@ func (ar *allocRunner) initRunnerHooks() {
 	// directory path exists for other hooks.
 	ar.runnerHooks = []interfaces.RunnerHook{
 		newAllocDirHook(hookLogger, ar.allocDir),
-		newDiskMigrationHook(hookLogger, ar.prevAllocWatcher, ar.allocDir),
+		newPreemptionHook(hookLogger, ar.prevAllocWatcher),
+		newDiskMigrationHook(hookLogger, ar.prevAllocMigrator, ar.allocDir),
```

Comment:
Not sure why the disk migration hook needs a prevAllocMigrator rather than prevAllocWatcher. Especially since here https://github.com/hashicorp/nomad/pull/4963/files#diff-c1417b3fea49cf8061386194ec47f04bR154 it looks like the migrator is really a watcher.

Reply:
They're the same underlying structs for a local and remote watcher, but it's a separate Interface for the purpose of clarity in the hooks themselves, and to avoid two fns that just error out on the group watcher.